### PR TITLE
Clean up and optimize printing screen

### LINKED ISSUE #1182: [FR] CHAMBER temperature
Please be able to display BED and CHAMBER temperature alternately.

### LINKED ISSUE #2085: [Q] (Is there a way to show the chamber temp on tft50 in touch mode. Using the latest version of firmware
**Your question or details you need. Please describe.**
Is there a way to show the chamber temps and heating from the main info screen on tft50. I can only see it in marlin mode

**specify the device variant related to your question.** btt tft50v3

**Additional context**
Add any other context or screenshots about the question here.

### PR DESCRIPTION
**Pull request by guruathwal**
- Clean up and reduce RAM usage in the Printing screen.
- Optimize the Printing screen to load faster.
- Prevent unnecessary status icon redraw in the Printing screen.
- Status icons on the Printing screen are now clickable after the print is finished.
- Clicking on the Layer/Z-layer status icon now opens the Babystep menu.
-  Misc. clean-up and reduce global variables.
-  Add chamber temperature status in status screen and printing screen (#1182). The bed heating icon will toggle between bed and chamber status If chamber heating is available.

Resolves #1182, Resolves #2085

### CODE REVIEW

**Comment by digant73** [comment]
@Gurmeet hi, in one of my next PR I planned to implement FR #1182. Your PR is already optimizing/changing the Printing menu etc... So maybe you could also integrate FR #1182. It should be quite easy unless you think that FR is not good to be implemented (I think once chamber is displayed on the notification bar it makes also more sense to provide it in the Status Screen and Printing menus). Please let me know your opinion

**Comment by guruathwal** [comment]
Hi @digant73, Yes I'll be able to implement it.

File: TFT/src/User/Menu/MeshEditor.c [hunk 1]
```diff
@@ -324,8 +317,10 @@ static inline bool processKnownDataFormat(char *dataRow)
 
 void meshSaveCallback(void)
 {
-  if (meshSaveCallbackPtr != NULL)
-    meshSaveCallbackPtr();
+ if (infoMachineSettings.leveling == BL_UBL)
+    menuUBLSave();
+  else if(infoMachineSettings.leveling != BL_DISABLED)
```
Comment (digant73) [inline TFT/src/User/Menu/MeshEditor.c:322]: "else if (". instead of "else if("

**Comment by oldman4U** [comment]
Gentlemen: @radek8 @AntoszHUN @bepstein111 

printing_chamber.bmp and status_chamber.bmp is required for this PR. Could you please generate them.

Thank you

PS: Ben, I believe there is one Icon missing in your package. Please check

**Comment by bepstein111** [comment]
Whoops! Adding calibration.bmp in (which I was missing) and will add the printing_chamber and status_chamber as soon as I see it's been added to the Red theme...I'm a terrible graphic designer, I just know how to work photoshop....

**Comment by AntoszHUN** [comment]
> Whoops! Adding calibration.bmp in (which I was missing) and will add the printing_chamber and status_chamber as soon as I see it's been added to the Red theme...I'm a terrible graphic designer, I just know how to work photoshop....

What's wrong now? I do not understand.

**Comment by oldman4U** [comment]
Ben missed something 😁

**Comment by AntoszHUN** [comment]
> Ben missed something 😁

I see a problem with the photoshopped icon. I have a little more time now. I'll do it right. It was just firefighting.

**Comment by bepstein111** [comment]
Oh jeez what'd i miss....

**Comment by AntoszHUN** [comment]
> 
> 
> Oh jeez what'd i miss....
> @AntoszHUN, nothing's wrong I don't think...did you already add printing_chamber and status_chamber to your Red theme? I didn't see it but I could have missed it.
> Is there something wrong with the calibration.bmp files? All I meant by my last sentence is that I'm not a good person for creating the icons (because I literally lack the talent required to do so), but if you create them for your Red theme, I can do the photoshopping to make the Mono theme so you don't have to maintain both.

I'm putting up the new missing icons tonight. Unfortunately I work a lot and I can deal with it in the evening, at night.

**Comment by MarkusThur** [comment]
@guruathwal I very much appreachiate the adding of heated chamber. Will try to test this later. Thanks a lot already

**Pull request merged**